sdk-extensions/autoconfigure/src/testFullConfig/java/io/opentelemetry/sdk/autoconfigure/EndpointConfigurationTest.java

```diff
@@ -0,0 +1,184 @@
+import io.opentelemetry.sdk.metrics.data.AggregationTemporality;
+import io.opentelemetry.sdk.metrics.data.LongPointData;
+import io.opentelemetry.sdk.metrics.data.LongSumData;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.testing.trace.TestSpanData;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import java.util.Collections;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingDeque;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+@SuppressWarnings("InterruptedExceptionSwallowed")
+class EndpointConfigurationTest {
+
+  private static final BlockingQueue<ExportTraceServiceRequest> otlpTraceRequests =
+      new LinkedBlockingDeque<>();
+  private static final BlockingQueue<ExportMetricsServiceRequest> otlpMetricsRequests =
+      new LinkedBlockingDeque<>();
+
+  @RegisterExtension
+  public static final ServerExtension server =
+      new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+          sb.service(
+              GrpcService.builder()
+                  // OTLP spans
+                  .addService(
+                      new TraceServiceGrpc.TraceServiceImplBase() {
+                        @Override
+                        public void export(
+                            ExportTraceServiceRequest request,
+                            StreamObserver<ExportTraceServiceResponse> responseObserver) {
+                          otlpTraceRequests.add(request);
+                          responseObserver.onNext(ExportTraceServiceResponse.getDefaultInstance());
+                          responseObserver.onCompleted();
+                        }
+                      })
+                  // OTLP metrics
+                  .addService(
+                      new MetricsServiceGrpc.MetricsServiceImplBase() {
+                        @Override
+                        public void export(
+                            ExportMetricsServiceRequest request,
+                            StreamObserver<ExportMetricsServiceResponse> responseObserver) {
+                          if (request.getResourceMetricsCount() > 0) {
+                            otlpMetricsRequests.add(request);
+                          }
+                          responseObserver.onNext(
+                              ExportMetricsServiceResponse.getDefaultInstance());
+                          responseObserver.onCompleted();
+                        }
+                      })
+                  .useBlockingTaskExecutor(true)
+                  .build());
+        }
+      };
+
+  @BeforeEach
+  void setUp() {
+    otlpTraceRequests.clear();
+    otlpMetricsRequests.clear();
+    GlobalOpenTelemetry.resetForTest();
```

Comment:
Does it makes sense to do this in the `tearDown` rather than `setUp`? Or maybe both places would be safest?

Reply:
added